### PR TITLE
Draft a new Setup script to create or get users on canton

### PR DESCRIPTION
**Pull request by mlachkar-da**
The goal is to make easier to run Daml script on Canton.

### CODE REVIEW

File: templates/create-daml-app/daml/Setup.daml [hunk 1]
```diff
@@ -3,45 +3,63 @@ module Setup where
...
+participantNamesMap : M.TextMap ParticipantName
+participantNamesMap = M.fromList $ map (\ParticipantDatails{alias, name}  -> (alias, name)) listParticipantDetails
 
 -- | Create a public party, then create three test users.
 setup : Script ()
 setup = do
-  public <- createPublic
-  let aliases = ["Alice", "Bob", "Charlie"]
-  forA_ aliases $ \alias -> createTestUser $ TestUser alias public
+  let aliases = [("Alice", "participant1"), ("Bob", "participant2")]
```
[the remark this answers is not in the document]
Comment (cocreature) [reply templates/create-daml-app/daml/Setup.daml:31]: I think I don’t understand how your distinction between participant alias and participant name maps to the current Daml Script API.

If we look at the `participants.json` format, the only thing users specify is what I’d call an alias, i.e., the key used in the `participants` map. That alias is then also what you use in `ParticipantName` in the `*On` functions.

The actual participant id (which is the `ParticipantAlias::…` identifier in Canton afaik) is not exposed to users anywhere within Daml Script.

Are you suggesting that we change the meaning of `ParticipantName` to be the participant id? How would the format in `participants.json` change?

[the remark this answers is not in the document]
Comment (mlachkar-da) [reply templates/create-daml-app/daml/Setup.daml:31]: So actually the code doesn't work with `ParticipantID` (from my tests) and requires the key. So maybe if we can verify that it works or not and then fix it, we may not need listPartipantDetails and just add a Daml script that shows how to run a script on Canton.

[the remark this answers is not in the document]
Comment (cocreature) [reply templates/create-daml-app/daml/Setup.daml:31]: Sounds good, it should just work™ if you use the alias (i.e. the key in the map) everywhere.

File: templates/create-daml-app/daml/Setup.daml [hunk 2]
```diff
@@ -3,45 +3,63 @@ module Setup where
...
+import qualified DA.TextMap as M
 import Daml.Script
 
 -- | A test user for the create-daml-app network.
 data TestUser = TestUser with
   alias : Text
   public : Party
+  participantAlias: Text
+
+data ParticipantDatails = ParticipantDatails with
```
Comment (thibault-da) [inline templates/create-daml-app/daml/Setup.daml:15]: Typo? ParticipantD**e**tails

Comment (cocreature) [reply templates/create-daml-app/daml/Setup.daml:15]: yeah, we don’t expect to merge this though so doesn’t really matter.

**Pull request closed without merging**